logstash-core/src/main/java/org/logstash/instrument/metrics/gauge/LazyDelegatingGauge.java

```diff
@@ -38,10 +35,8 @@ public LazyDelegatingGauge(final String key) {
      *
      * @param key          The key for this metric
      * @param initialValue The initial value for this {@link GaugeMetric}, may be null
-     * @deprecated - there are no plans to replace this
      */
-    @Deprecated
-    public LazyDelegatingGauge(String key, Object initialValue) {
```

Comment:
Why do we still need this?

Reply:
This constructor, or this class ?

(Assuming you mean this class)
This is the backing implementation used by the plugins that all them to continue to duck type. For example from the beats input (ruby side):
```ruby
@metric.gauge(:peak_connections, @peak_connection_count.value)
```
Will find its way to here, and we lazily implement the correctly typed Gauge.